app/controllers/ems_common.rb

```diff
@@ -832,6 +871,40 @@ def get_form_vars
     set_verify_status
   end
 
+  def set_record_creds(ems)
+    creds = {}
+    creds[:default] = {
+      :userid   => @edit[:new][:default_userid],
+      :password => @edit[:new][:default_password]
+    } unless @edit[:new][:default_userid].blank?
+
+    if ems.supports_authentication?(:metrics) && !@edit[:new][:metrics_userid].blank?
```

Comment:
I know you're just moving code around but... `.present?` instead of `!....blank?` to get rid of the negation.

Also, the hash access is much faster than anything `supports_authentication?` is doing so the conditionals should be swapped... also it seems to make more sense to only check if the ems supports an auth if it has that auth's userid.